src/compilation/config.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    content_encoding::{
+        ContentTypeCheckType, ContentTypeConverterType, DEFAULT_CONTENT_ENCODING_CHECK,
+        DEFAULT_CONTENT_ENCODING_CONVERT,
+    },
+    media_type::{MediaTypeCheckType, DEFAULT_MEDIA_TYPE_CHECK},
+    schemas,
+};
+use serde_json::Value;
+use std::{collections::HashMap, fmt};
+
+#[allow(missing_docs)]
+#[derive(Clone, Default)]
+pub struct JSONSchemaCompileConfig {
```

Comment:
What do you think about `CompilationConfig` instead?

Reply:
Totally in for this ... the proposed name is far from being communicative and is unreasonably long.